ui/webui/resources/wasm/candle_embedding_gemma/src/lib.rs

```diff
@@ -3,4 +3,986 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at https://mozilla.org/MPL/2.0/. */
 
+// Directly downloaded from https://github.com/brave-experiments/on-device-experiments/blob/main/gemma_wasm/src/lib.rs
+// Embedding Gemma inference implementation,
+// modified from 'https://github.com/huggingface/text-embeddings-inference/blob/main/backends/candle/src/models/gemma3.rs'
+
+use candle_core::{DType, Device, IndexOp, Tensor, D};
+use candle_nn::{Embedding, Module, VarBuilder};
+use serde::Deserialize;
+use tokenizers::Tokenizer;
 use wasm_bindgen::prelude::*;
+
+// Console logging macro
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(js_namespace = console)]
+    pub fn log(s: &str);
+}
+
+#[macro_export]
+macro_rules! console_log {
+    ($($t:tt)*) => ($crate::log(&format_args!($($t)*).to_string()))
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct NTKScaling {
+    pub factor: f32,
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+#[serde(tag = "type", rename_all = "kebab-case")]
+pub enum RopeScaling {
+    Ntk(NTKScaling),
+}
```

Comment:
I assume the enum with only 1 variant is just for the purpose of deserialization?